### ISSUE TITLE
chips: accessibility improvements

### ISSUE THREAD

**Issue by marcysutton** [angular/material#2618]
There are a few issues in mdChips that make them hard to use with a screen reader, including:
- The input needs a label enforced - right now it's empty if no placeholder
- When the input is focused, it should announce the label, not "Press delete to remove this chip"
- Chips need to be uniquely labeled along with the delete hint text. Ex: when a chip is focused, it should say: "Apple. Press delete to remove this chip".
- Need to verify chips can be removed from touch devices with screen readers running.

**Comment by marcysutton** [comment]
This is complicated by the fact that there are three different ways a Chips component gets a native input, which is the thing requiring a label:
1. Using an autocomplete, which has its own `<input>`
1. Manually adding an `<input>`
1. Chips dynamically adds an `<input>`

If there is a placeholder on the parent, we can copy that text to the child input IF it doesn't already have `aria-label` (likely when manually adding an input). If the parent has `aria-label` already, that should be moved to the input instead. 

Bonus points for supporting `aria-labelledby`. But I can't even get a reliable `aria-label` in all three scenarios.

**Comment by marcysutton** [comment]
An accessible chips component would look like this: 
```html
<md-chips ng-model="ctrl.roFruitNames" tabindex="-1">
    <md-chips-wrap ng-if="!$mdChipsCtrl.readonly || $mdChipsCtrl.items.length > 0" 
         ng-keydown="$mdChipsCtrl.chipKeydown($event)" 
         ng-class="{ 'md-focused': $mdChipsCtrl.hasFocus() }" class="md-chips">

        <md-chip ng-repeat="$chip in $mdChipsCtrl.items" index="0" 
                       ng-class="{'md-focused': $mdChipsCtrl.selectedChip == $index}" 
                       class="md-chip">          

            <div class="md-chip-content" tabindex="-1" 
                ng-focus="!$mdChipsCtrl.readonly && $mdChipsCtrl.selectChip($index)">
                <md-chip-template>
                    <strong>Apple</strong>
                    <em>(fruit)</em>
                </md-chip-template>
          </div>          

          <div class="md-chip-remove-container"> 
              <button class="md-chip-remove" 
                    ng-click="$mdChipsCtrl.removeChipAndFocusInput($$replacedScope.$index)" 
                    type="button" 
                    aria-hidden="true" tabindex="-1"
                    id="uniqueButtonID">        
                    <md-icon md-svg-icon="md-close" aria-hidden="true"></md-icon>
            </button>
          </div>
        </md-chip>

        <div class="md-chip-input-container">        
               <input ng-model="$mdChipsCtrl.chipBuffer" 
                         ng-focus="$mdChipsCtrl.onInputFocus()" 
                         ng-blur="$mdChipsCtrl.onInputBlur()" 
                         ng-keydown="$mdChipsCtrl.inputKeydown($event)"
                         aria-label="Some label"
                         placeholder="Placeholder"
                         aria-describedby="deleteHintText">
               <div id="deleteHintText" style="display: none;">Press delete to remove {{ lastChip.name }} </div>
        </div>          
    </md-chips-wrap>
</md-chips>
```

The major differences are:
 * The input has a label.
 * The input has `aria-describedby` pointing to a new hint text node telling a screen reader user that the delete key will remove the last chip added (with a unique name).
 * The remove button doesn't need offscreen text since it has `aria-hidden` and is not in the tab order.

**Comment by marcysutton** [comment]
@ThomasBurleson `aria-label` isn't making its way to the input in the chips demo that uses autocomplete. The static chips demo also throws a warning. Can you take a look at those?

**Comment by ilanbiala** [comment]
@marcysutton @ThomasBurleson any progress on this?

**Comment by ThomasBurleson** [comment]
@ilanbiala - The team is focused and working very hard on features and issue resolution for v1.0. Issue comments posted requesting progress/status updates are not helpful.

**Comment by ilanbiala** [comment]
@ThomasBurleson this is marked for 0.11.0, which is the next release and the latest commit was made over a month ago, I'm trying to find out if any progress has been made that hasn't been reported or if there are issues and maybe someone from the community can help.

**Comment by topherfangio** [comment]
I am scheduling some other chips issues for RC6 and will try to look at this then.

**Comment by biomassives** [comment]
This message was created automatically by mail delivery software.

A message that you sent could not be delivered to one or more of its
recipients. This is a temporary error. The following address(es) deferred:

  acmeideal@gmail.com
    Domain biomassiv.es has exceeded the max emails per hour (621/500 (124%)) allowed.  Message will be reattempted later

------- This is a copy of the message, including all the headers. ------
Received: from github-smtp2-ext2.iad.github.net ([192.30.252.193]:57984 helo=github-smtp2b-ext-cp1-prd.iad.github.net)
	by ps5.websitehostserver.net with esmtps (TLSv1.2:ECDHE-RSA-AES256-GCM-SHA384:256)
	(Exim 4.86_1)
	(envelope-from <noreply@github.com>)
	id 1avsdM-000Lmu-Ok
	for greg@biomassiv.es; Thu, 28 Apr 2016 15:34:00 -0500
Date: Thu, 28 Apr 2016 13:32:28 -0700
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=github.com;
	s=pf2014; t=1461875548;
	bh=heIg09QZYZqOxr862qH67OIHkmrrVEP3NdizkgXNAWo=;
	h=From:Reply-To:To:Cc:In-Reply-To:References:Subject:List-ID:
	 List-Archive:List-Post:List-Unsubscribe:From;
	b=WMj7Zv6VZwuKkugMgLWxL5syebImoZ3rnEj4EdmICooEMS71GHUb8nPT5g/l8OJuA
	 GYPOzAjxvS0p/yhZA0e1SkVCjJ6ctiCTA/qYllg9tPXW3xYq7+nkQVt68+fL7pJeE0
	 SAgD5GkITuh3KO7RlDKKdZAuGYOJCI01Q9v5Zj1o=

**Comment by Splaktar** [comment]
Chips currently have a lot of accessibility issues. This issue is helpful as it indicates some possible solutions, so I am re-opening it.

**Comment by Splaktar** [comment]
I've posted PR https://github.com/angular/material/pull/11422 to address these issues and more. I'd appreciate some assistance with the review 😄
